### PR TITLE
Improve C# code practice

### PR DESCRIPTION
**Pull request by vadi2**
#### Brief overview of PR changes/additions
Made fields that can be private private, readonly ones readonly, and renamed a few things to C# convention (even though it feels weird)
#### Motivation for adding to Hammer
Better C# code

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -236,7 +236,7 @@ public ValidationResult DotnetResult
     
     private CancellationTokenSource _validatorCancellationSource;
     
-    private List<Process> _validatorProcesses = new List<Process>();
+    private readonly List<Process> _validatorProcesses = new List<Process>();
```
Comment (pieter-edelman-nictiz) [inline Program.cs]: This confused me a bit, but yeah, this is totally correct :)

Comment (vadi2) [reply Program.cs]: I had to think about it too!

File: Program.cs [hunk 2]
```diff
@@ -51,8 +51,8 @@ public void Dispose()
     private readonly IResourceResolver _coreSource = new CachedResolver(ZipSource.CreateValidationSource());
 
     private IResourceResolver _combinedSource;
-    
-    Regex cleanFhirPath = new Regex(@"([^\(]+)", RegexOptions.Compiled);
+
+    private readonly Regex _cleanFhirPath = new Regex(@"([^\(]+)", RegexOptions.Compiled);
```
[the remark this answers is not in the document]
Comment (vadi2) [reply Program.cs]: Just so it's compiled once, really.

[the remark this answers is not in the document]
Comment (pieter-edelman-nictiz) [reply Program.cs]: What about making it `static` in `SanitizeLocation()`?

[the remark this answers is not in the document]
Comment (vadi2) [reply Program.cs]: Doesn't seem to be a thing in C#? https://stackoverflow.com/questions/4525654/c-static-variable-in-a-static-method#4525660

I didn't know this either.

[the remark this answers is not in the document]
Comment (pieter-edelman-nictiz) [reply Program.cs]: That's interesting ...

MS does provide guidance on caching regexes however: https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices#static-regular-expressions. In short: use a static call to Regex.Match(input, pattern) and all should be fine.

[the remark this answers is not in the document]
Comment (vadi2) [reply Program.cs]: Okay, fixed, I think? I don't have a working Java validator anymore due to https://github.com/hapifhir/org.hl7.fhir.core/issues/57...

**Pull request merged**